pyirf/io/tests/test_gadf.py

```diff
@@ -85,3 +84,33 @@ def test_psf_table():
             # unlike how it handles effective area and edisp
             # see https://github.com/gammapy/gammapy/issues/3025
             assert u.allclose(psf, psf3d.psf_value.T, atol=1e-16 / u.sr)
+
+
+# gammapy uses inconsistent axis order, should be fixed before gammapy 1.0
+# see https://github.com/gammapy/gammapy/issues/2067
+# TODO: remove xfail when this is fixed in gammapy and bump the required gammapy version
+@pytest.mark.xfail
+def test_background_2d():
+    '''Test our background hdu is readable by gammapy'''
+    pytest.importorskip('gammapy')
+    from pyirf.io import create_background_2d_hdu
+    from gammapy.irf import Background2D
+
+    e_bins = np.geomspace(0.1, 100, 31) * u.TeV
+    fov_bins = [0, 1, 2, 3] * u.deg
+    background = np.column_stack([
+        np.geomspace(1e9, 1e3, 3),
+        np.geomspace(0.5e9, 0.5e3, 3),
+        np.geomspace(1e8, 1e2, 3),
+    ]) * u.Unit('TeV-1 s-1 sr-1')
+
+    for point_like in [True, False]:
+        with tempfile.NamedTemporaryFile(suffix='.fits') as f:
+            hdu = create_background_2d_hdu(background, e_bins, fov_bins)
+
+            fits.HDUList([fits.PrimaryHDU(), hdu]).writeto(f.name)
+
+            # test reading with gammapy works
+            bg2d = Background2D.read(f.name, 'BACKGROUND')
+
+            assert u.allclose(background, bg2d.data.data, atol=1e-16 * u.Unit('TeV-1 s-1 sr-1'))
```

Comment:
I think for the `Background2D` class in Gammapy the axis order is equivalent, so you can actually implement the test, no?

```python3
from astropy import units as u
from gammapy.irf import Background2D

energy = [0.1, 10, 1000] * u.TeV
offset = [0, 1, 2, 3] * u.deg
data = np.zeros((2, 3)) * u.Unit("s-1 MeV-1 sr-1")
data.value[1, 0] = 2
data.value[1, 1] = 4

bkg = Background2D(
    energy_lo=energy[:-1],
    energy_hi=energy[1:],
    offset_lo=offset[:-1],
    offset_hi=offset[1:],
    data=data,
)

print(bkg.data.data.shape)
print([ax.name for ax in bkg.data.axes])
```
Which prints:
```
(2, 3)
['energy', 'offset']
```

Reply:
The test here is meant to verify that the file / hdu can be read. So creating the in-memory version does not really help here.

Reply:
We opted to in-memory have the numpy convention for axis order like in the GADF doc page, which means we transpose when reading and writing from / to fits.